transmart-core-db/src/main/groovy/org/transmartproject/db/multidimquery/SubjectObservationsByStudyConceptsTableView.groovy

```diff
@@ -0,0 +1,236 @@
+            switch (metadata?.type) {
+                case NUMERIC:
+                    if (value instanceof Number) return value
+                    String label = value == null ? getMissingValueLabel(hValue) : value
+                    if (labelsToValues.containsKey(label)) return labelsToValues[label]
+                    if (value == null) return null
+                    throw new UnexpectedException("${value} is not of a number type.")
+                case DATE:
+                    if (value == null) return null
+                    if (value instanceof Number) return toDate(value)
+                    throw new UnexpectedException("${value} can't be converted to the date type.")
+                case STRING:
+                    return value as String
+                default:
+                    return value
+            }
+        }
+
+        private Date toDate(Number value) {
+            new Date((value * 1000) as Long)
+        }
+
+        private ColumnMetadata computeColumnMetadata() {
+            new ColumnMetadata(
+                    type: STRING,
+                    measure: NOMINAL,
+                    description: originalColumn.label,
+                    width: 25,
+                    columns: 25,
+            )
+        }
+
+        private String getMissingValueLabel(HypercubeValue hValue) {
+            hValue.availableDimensions
+                    .find { it.name == MISSING_VALUE_MODIFIER_DIMENSION_NAME }
+                    .with { naDim -> hValue[naDim] }
+        }
+
+        ColumnMetadata parseColumnMetadata(json, String description) {
+            if (!json) return null
+            new ColumnMetadata(
+                    type: json.type?.toUpperCase() as ColumnDataType,
+                    measure: json.measure?.toUpperCase() as Measure,
+                    description: description,
+                    width: json.width,
+                    decimals: json.decimals,
+                    valueLabels: json.valueLabels?.collectEntries { key, value -> [key as Integer, value] } ?: [:],
+                    missingValues: json.missingValues?.collect { it as Integer } ?: [],
+                    columns: json.columns
+            )
+        }
+    }
+
+    private static String getVariableName(HypercubeDataColumn originalColumn) {
+        Study study = originalColumn.getDimensionElement(DimensionImpl.STUDY)
+        org.transmartproject.db.i2b2data.ConceptDimension conceptDimension =
+                originalColumn.getDimensionElement(DimensionImpl.CONCEPT)
+
+        def studyMetadata = study.studyBlobAsJson()
+        if (studyMetadata
+                && studyMetadata.conceptToVariableName
+                && conceptDimension.conceptCode in studyMetadata.conceptToVariableName) {
+            return studyMetadata.conceptToVariableName[conceptDimension.conceptCode]
+        } else {
+            conceptDimension.conceptCode
+        }
+    }
+}
+
+enum ColumnDataType {
```

Comment:
I would prefer these type to be in core, so that they can be reused in the REST API for exporting the concept metadata. E.g., when exporting to the data showcase.